test/e2e/utils_test.go

```diff
@@ -489,3 +492,14 @@ func setEnv(p setEnvParams) error {
 	}
 	return nil
 }
+
+// dumpToTempFile dumps the contents of the reader to a temporary file and returns the path to the file.
+func dumpToTempFile(t *testing.T, reader io.Reader) string {
+	t.Helper()
+	file, err := os.CreateTemp("", "")
+	require.NoError(t, err)
+	defer file.Close()
```

Comment:
Why not use `t.TempDir()`? It automatically removes the file during `t.Cleanup()`